lib/usd/utils/MayaTransformAPI.h

```diff
@@ -0,0 +1,263 @@
+  /// \return the rotatePivot at the specified time code
+  MAYA_USD_UTILS_PUBLIC
+  GfVec3f rotatePivot(const UsdTimeCode& time = UsdTimeCode::Default()) const;
+
+  /// \brief  returns the scalePivotTranslate at the specified time code
+  /// \param  time the time value to query the UsdGeomXform
+  /// \return the scalePivotTranslate at the specified time code
+  MAYA_USD_UTILS_PUBLIC
+  GfVec3f scalePivotTranslate(const UsdTimeCode& time = UsdTimeCode::Default()) const;
+
+  /// \brief  returns the rotatePivotTranslate at the specified time code
+  /// \param  time the time value to query the UsdGeomXform
+  /// \return the rotatePivotTranslate at the specified time code
+  MAYA_USD_UTILS_PUBLIC
+  GfVec3f rotatePivotTranslate(const UsdTimeCode& time = UsdTimeCode::Default()) const;
+
+  /// \brief  returns true if the parent transforms affect this UsdGeomXform
+  /// \return true if this xform is influenced by the parent transform
+  MAYA_USD_UTILS_PUBLIC
+  bool inheritsTransform() const;
+
+  /// \brief  evaluates the matrix at the specified time code
+  /// \param  time the time at which to evaluate the transform
+  /// \return the local speace matrix for this prim
+  MAYA_USD_UTILS_PUBLIC
+  GfMatrix4d asMatrix(const UsdTimeCode& time = UsdTimeCode::Default()) const;
+
+  /// \brief  sets the current transform from a matrix at the specified time. The matrix will be decomposed
+  ///         and set as TRS components. 
+  /// \param  matrix the new local space matrix for the prim
+  /// \param  time the time code at which to set the transform
+  MAYA_USD_UTILS_PUBLIC
+  void setFromMatrix(const GfMatrix4d& matrix, const UsdTimeCode& time = UsdTimeCode::Default());
+
+  /// \brief  checks to see if the transform can be manipulated by this class. 
+  /// \return If this method returns true, then this class can manipulate the transform. 
+  ///         If this returns false, we cannot modify the transform (i.e. it doesn't match the maya xform profile,
+  ///         or the prim contains a single matrix transform, but convertMatrixOpToComponentOps was set to false)
+  operator bool () const 
+    { return isValid(); }
+
+  /// \brief  checks to see if the transform can be manipulated by this class. 
+  /// \return If this method returns true, then this class can manipulate the transform. 
+  ///         If this returns false, we cannot modify the transform (i.e. it doesn't match the maya xform profile,
+  ///         or the prim contains a single matrix transform, but convertMatrixOpToComponentOps was set to false)
+  bool isValid() const 
+    { return bool(m_prim); }
+
+  /// \brief  returns the API this prim is compatible with
+  /// \return the transform API that was found
+  TransformAPI api() const
+    { return m_api; }
+
+protected:
+  void convertMatrixOpToComponentOps(const UsdGeomXformOp& op);
+  bool matchesMayaTrasformProfile(const std::vector<UsdGeomXformOp>& orderedOps);
```

Comment:
Typo: "matchesMayaTrasformProfile" --> "matchesMayaTransformProfile".

Also awkward that this is not const.